test/mpi/dtpools/src/dtpools_attr.c

```diff
@@ -24,6 +24,33 @@
         }                                                               \
     } while (0)
 
+static void generate_random_blklens(DTP_pool_s dtp, int numblks, int *blklens, MPI_Aint real_count)
+{
+    MPI_Aint count = real_count;
+    DTPI_pool_s *dtpi = dtp.priv;
+
+    /* find a random divisor of count by randomly choosing which
+     * factors of count to use */
+    int total_blklen = 1;
+    while (count > 1) {
+        if (DTPI_low_count(count) == 1) {
+            total_blklen *= (DTPI_rand(dtpi) % 2) ? 1 : count;
+            break;
+        } else {
+            /* randomly pick or ignore this divisor */
+            MPI_Aint lowcount = DTPI_low_count(count);
+            total_blklen *= (DTPI_rand(dtpi) % 2) ? 1 : lowcount;
+            count /= lowcount;
+        }
+    }
+
+    for (int i = 0; i < numblks; i++)
+        blklens[i] = 0;
+
+    for (int i = 0; i < total_blklen; i++)
+        blklens[DTPI_rand(dtpi) % numblks]++;
+}
+
```

Comment:
@pavanbalaji This algorithm is different from generate `numblks-1` dividers directly. By randomly add 1 to one of the block, the probability of have a 0 block is `[(numblks-1)/numblks]^total_blklen`, even with 10 blocks and 100 total, `0.9^100` is `0.000026`, i.e. very small. By randomly picking `numblks-1` divider, according to https://en.wikipedia.org/wiki/Birthday_problem, is `~0.39`, i.e. very significant. When `total_blklen` is very big, your algorithm will result in very uniform blocks, but the other algorithm will result in quite "random" block sizes.

Reply:
Talking to some friend and here is another algorithm: simply generate `numblks` random numbers then scale down (or up) to the total. The last one may need padding of plus or minus one.

With this algorithm, the distribution of each `blklen` is uniform (from 0 to the scaled upperbound). In the case of randomly adding `1`, the distribution of each `blklen` is gaussian. I am not exactly sure yet what is the distribution randomly picking dividers would result.